packages/ui-components/src/components/common/loading/ProgressBar.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div :class="[containerStyle, loading ? 'opacity-100' : 'opacity-0']">
+    <div
+      :class="[progress ? '' : 'swoosher top-0', barStyle]"
+      :style="widthStyle"
+    ></div>
+  </div>
+</template>
+<script setup lang="ts">
+import { computed } from 'vue'
+const props = defineProps<{
```

Comment:
Please add a Story for this new component, like all of our other ui-components. Even if it's just a single story rendering the progress bar.